client/state/billing-data/util.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+import { moment } from 'i18n-calypso';
+
+/**
+ * Parses the date within a transaction.
+ * Returns a copy of the updated transaction.
+ *
+ * @param  {Object}  transaction  Transaction object
+ * @return {Object}               Updated transaction with parsed date
+ */
+export const parseDate = ( transaction ) => {
+	return assign( {}, transaction, {
```

Comment:
Optional, but you can also do:
```
return { ...transaction, date: moment( transaction.date ).toDate() };
```

Also is [Date.parse](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse) too unreliable for our date string?